spirv_hlsl.cpp

```diff
@@ -1079,6 +1095,13 @@ void CompilerHLSL::emit_builtin_variables()
 			else
 				statement("static ", type, " ", builtin_to_glsl(builtin, storage), ";");
 		}
+
+		// SampleMask can be both in and out with sample builtin, in this case we have already
+		// declared the input variable and we need to add the output one now.
+		if (builtin == BuiltInSampleMask && storage == StorageClassInput && active_output_builtins.get(i))
+		{
+			statement("static ", type, " ", builtin_to_glsl(builtin, StorageClassOutput), ";");
```

Comment:
This likely is what is failing AppVeyor build for MSVC 2013 and 2015. You need to use `this->` to access member functions to workaround these bugs.

Reply:
It passed the AppVeyor build now.